matplotlibrc.template

```diff
@@ -1,46 +1,83 @@
 #### MATPLOTLIBRC FORMAT
 
 ## This is a sample matplotlib configuration file - you can find a copy
```

Comment:
Just as a remark: The whole format description would better be placed at https://matplotlib.org/devdocs/tutorials/introductory/customizing.html#the-matplotlibrc-file. In the rc file itself, a 1-2 sentence description and the above link should be sufficient. This declutters the rc file. Additionally, the rendered docs are easier to read and search.

But I'm happy to leave that to another PR.

Reply:
I see. I can do this in another PR or with one or two more commits to declutters the RC file.

Reply:
I think it's better done in a separate PR. This one is already quite large to review.

Reply:
Yes, it is. I will open another PR later on 😄